app/src/main/java/com/vultisig/wallet/presenter/home/HomeViewModel.kt

```diff
@@ -1,26 +1,31 @@
 package com.vultisig.wallet.presenter.home
 
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.vultisig.wallet.data.on_board.db.VaultDB
 import com.vultisig.wallet.models.Vault
+import com.vultisig.wallet.ui.components.reorderable.utils.ItemPosition
 import com.vultisig.wallet.ui.navigation.Destination
 import com.vultisig.wallet.ui.navigation.Navigator
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 
 @HiltViewModel
 internal class HomeViewModel @Inject constructor(
-    vaultDB: VaultDB,
+    private val vaultDB: VaultDB,
     private val navigator: Navigator<Destination>
 ) : ViewModel() {
-    val vaults: MutableLiveData<List<Vault>> = MutableLiveData(listOf())
+
+    var vaults = MutableStateFlow<List<Vault>> (emptyList())
 
     init {
-        vaults.postValue(vaultDB.selectAll())
+        vaults.update {
+            vaultDB.selectAll()
```

Comment:
As in [previous PR](https://github.com/vultisig/vultisig-android/pull/134#discussion_r1605977378), the update's lambda can be reevaluated more than once, so consider moving calculations out of it.

Reply:
Also, if you don't care for the current value of state flow, using `Stateflow.value = value` may be a bit more optimal